source/common/response_map/response_map.cc

```diff
@@ -0,0 +1,204 @@
+#include "common/response_map/response_map.h"
+
+#include <string>
+#include <vector>
+
+#include "common/access_log/access_log_impl.h"
+#include "common/common/enum_to_int.h"
+#include "common/config/datasource.h"
+#include "common/formatter/substitution_format_string.h"
+#include "common/formatter/substitution_formatter.h"
+#include "common/http/header_map_impl.h"
+#include "common/http/utility.h"
+
+#include "envoy/api/api.h"
+
+namespace Envoy {
+namespace ResponseMap {
+
+class BodyFormatter {
+public:
+  BodyFormatter()
+      : formatter_(std::make_unique<Envoy::Formatter::FormatterImpl>("%LOCAL_REPLY_BODY%")),
+        content_type_(Http::Headers::get().ContentTypeValues.Text) {}
+
+  BodyFormatter(const envoy::config::core::v3::SubstitutionFormatString& config, Api::Api& api)
+      : formatter_(Formatter::SubstitutionFormatStringUtils::fromProtoConfig(config, api)),
+        content_type_(
+            !config.content_type().empty()
+                ? config.content_type()
+                : config.format_case() ==
+                          envoy::config::core::v3::SubstitutionFormatString::FormatCase::kJsonFormat
+                      ? Http::Headers::get().ContentTypeValues.Json
+                      : Http::Headers::get().ContentTypeValues.Text) {}
```

Comment:
Wasn't there some discussion of commenting the initializer logic here, since the `Content-Type` stuff is kind of critical?

...oh, huh, wait. Will that screw up merging the stuff you backported this from?

Reply:
Yeah this is backported